docs/remote/containers-advanced.md

```diff
@@ -151,48 +151,65 @@ If you've already built the container and connected to it, run **Remote-Containe
+There are a few side effects of doing this you should be aware of:
 
-First, configure a volume mount for `~/.vscode-server` (and/or `~/.vscode-server-insiders` for VS Code Insiders). How you do this will depend on whether you specify an image, Dockerfile, or Docker Compose file in your `devcontainer.json` file.
+* Sharing the volume across multiple containers could have unintended consequences, so you should pick a unique name for each.
+* When you rebuild the container:
+  * New extensions added to devcontainer.json will **not** be automatically installed.
+  * Any `postCreateCommand` in `devcontainer.json` will **not run**.
+* Deleting the container will not automatically delete the named volume.
 
-**Dockerfile or image**:
+To create the named local volume, follow these steps:
 
-Replace `your-volume-name-goes-here` with a unique volume name for the container in `devcontainer.json` as follows:
+1. **If you are using a non-root user**, you'll need to ensure your Dockerfile creates `~/.vscode-server` and/or `~/.vscode-server-insiders` in the container.  If you do not do this, the folder will be owned by root and your connection will fail with a permissions issue. See [Adding a non-root user to your dev container](#adding-a-non-root-user-to-your-dev-container) for full details, but you can use this snippet in your Dockerfile to create the folders for a `vscode` user:
 
-```json
-"runArgs": ["-v","your-volume-name-goes-here:/root/.vscode-server"]
-```
+    ```Dockerfile
+    ARG USERNAME=vscode
+    RUN mkdir -p /home/$USERNAME/.vscode-server /home/$USERNAME/.vscode-server-insiders \
+        && chown $USERNAME:$USERNAME /home/$USERNAME/.vscode-server /home/$USERNAME/.vscode-server-insiders
+    ```
 
-...or for VS Code Insiders...
+2. Next, we'll configure a named volume mount for `~/.vscode-server` and `~/.vscode-server-insiders` in the container. How you do this will depend on whether you specify an image, Dockerfile, or Docker Compose file in your `devcontainer.json` file.
 
-```json
-"runArgs": ["-v","your-volume-name-goes-here:/root/.vscode-server-insiders"]
-```
+    **Dockerfile or image**:
 
-**Docker Compose**:
+    Add the following to `devcontainer.json`, replacing `/root` with the home directory of in the container if not root (e.g. `/home/vscode`) and `unique-vol-name-here` with a unique name for the volume:
 
-Update (or [extend](/docs/remote/containers.md#extending-your-docker-compose-file-for-development)) your `docker-compose.yml` with the following. Replace `your-volume-name-goes-here` with a unique volume name for the container.
+    ```json
+    "runArgs": [
+        "-v", "unique-vol-name-here:/root/.vscode-server",
+        // And/or for VS Code Insiders
+        "-v", "unique-vol-name-here-insiders:/.vscode-server-insiders",
```

Comment:
Would it be easier to only show the code for stable and add a note in the text that the same applies for the insiders folder?

Reply:
I had it that way, but then you need to say "also insiders" in each spot where this is needed, which actually ended up taking up more space and was easy to miss. This was a bit more concise and allows you to use both versions - cut and paste also covers both use cases so people don't have to think as much.